### PR TITLE
adding support for alerts/groups endpoint of alert manager

### PR DESCRIPTION
**Pull request by prashbnair**
This PR addresses issue https://github.com/prometheus-community/prom-label-proxy/issues/89 by adding support for alerts/groups endpoint in alert manager

### CODE REVIEW

File: injectproxy/alertgroups.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package injectproxy
+
+import (
+	"net/http"
+)
+
+func (r *routes) alertGroups(w http.ResponseWriter, req *http.Request) {
```
Comment (simonpasquier) [inline injectproxy/alertgroups.go:20]: since `alertGroups` and `listSilences` we can have a single method (maybe `enforceFilterParameter()`?).

Comment (prashbnair) [reply injectproxy/alertgroups.go:20]: Does it not make sense for it to be different methods and have them call into a common method?  This common method can have all the common code.  It makes the separation  of responsibilities clear

Comment (prashbnair) [reply injectproxy/alertgroups.go]: Done

**Review by simonpasquier — COMMENTED** [review]
could you add some unit tests please?

**Comment by prashbnair** [comment]
> could you add some unit tests please?
Done

**Review by simonpasquier — COMMENTED** [review]
The code change looks good to me. Could you just update the README.md to document the new endpoint?

**Comment by prashbnair** [comment]
> The code change looks good to me. Could you just update the README.md to document the new endpoint?

Done

**Review by simonpasquier — APPROVED** [review]
lgtm

**Pull request merged**